### PR TITLE
add changes

### PR DESCRIPTION
**Pull request by nirtiac**

### CODE REVIEW

File: vulcanai/plotters/visualization.py [hunk 1]
```diff
@@ -215,7 +216,7 @@ def _plot_reduction(x_transform, targets, label_map, title, save_path=None):
             "Display environment needed to plot, "
             "or set save_path=path/to/dir")
     else:
-        plt.show(False)
+        plt.show(block=False)
```
Comment (nirtiac) [inline vulcanai/plotters/visualization.py]: I believe that this will change the functionality: we want to have plots show up while the model is training, although this does depend heavily on a users' environment. Is there another way to get rid of whatever warning this is related to? https://stackoverflow.com/questions/458209/is-there-a-way-to-detach-matplotlib-plots-so-that-the-computation-can-continue/13361748

Comment (nirtiac) [reply vulcanai/plotters/visualization.py]: Repeat for all similar changes below. @rfratila may know more about the intended functionality as I wasn't involved in the writing of this.

File: vulcanai/tests/plotters/test_visualization.py [hunk 2]
```diff
@@ -7,6 +7,11 @@
 from vulcanai.datasets import tabulardataset
 import os
 from copy import deepcopy
+import matplotlib
+import tkinter
+matplotlib.use('TkAgg')
+import warnings
+
```
Comment (nirtiac) [inline vulcanai/tests/plotters/test_visualization.py]: why are there these new imports?

File: vulcanai/tests/plotters/test_visualization.py [hunk 3]
```diff
@@ -168,6 +172,7 @@ def test_display_pca(self, dnn_class):
 
     def test_display_tsne(self, dnn_class):
         """Test t-SNE displays and saves."""
+        warnings.filterwarnings('ignore', category=PendingDeprecationWarning)
```
Comment (nirtiac) [inline vulcanai/tests/plotters/test_visualization.py]: can we instead look for something to replace this that's not pending deprecation? we may find ourselves in trouble in the future and may as well figure out how to solve it now.

File: vulcanai/plotters/visualization.py (hunk 1 above)
Comment (rfratila) [reply vulcanai/plotters/visualization.py]: yes this line is to preserve the plot even once it is no longer training. it is compatible with running python interactively (i.e. in the command line you can run `python -i ...`) to keep the plots up even when the script finishes. More importantly, this is to keep the rest of the script from stalling so it doesn't force you to close the plot in order to continue running the rest of the script.

File: vulcanai/tests/models/test_metrics.py [hunk 4]
```diff
@@ -294,13 +293,11 @@ def test_get_auc(self, metrics):
...
         """Test that run_test returns values as expected."""
+        warnings.filterwarnings("ignore", category=sklearn.exceptions.UndefinedMetricWarning)
         num_items = 300
 
-        test_input = torch.Tensor(np.random.randint(0, 10,
-                                                    size=(num_items,
-                                                          *cnn_class.in_dim)))
-        test_target = torch.LongTensor(np.random.randint(0, 10,
-                                                         size=num_items))
+        test_input = torch.Tensor(np.random.randint(0, 10,size=(num_items,*cnn_class.in_dim)))
```
Comment (rfratila) [inline vulcanai/tests/models/test_metrics.py]: the length of this line may exceed to 79 char limit which is why it was formatted as it was

File: vulcanai/tests/models/test_metrics.py [hunk 5]
```diff
@@ -294,13 +293,11 @@ def test_get_auc(self, metrics):
 
     def test_run_test(self, metrics, cnn_class):
         """Test that run_test returns values as expected."""
+        warnings.filterwarnings("ignore", category=sklearn.exceptions.UndefinedMetricWarning)
```
Comment (rfratila) [inline vulcanai/tests/models/test_metrics.py]: is there another way to make sure the warning goes away without simply ignoring it? this applies to all the filter warnings?

**Comment by divyachandran-ds** [comment]
@nirtiac @rfratila 
I will work on all the above suggestions and get back to you.

**Comment by nirtiac** [comment]
Thanks Divya! :) I’m happy to answer any questions during the process.

On Jun 25, 2019, at 12:58 PM, divyachandran-ds <notifications@github.com<mailto:notifications@github.com>> wrote:


@nirtiac<https://github.com/nirtiac> @rfratila<https://github.com/rfratila>
I will work on all the above suggestions and get back to you.

File: vulcanai/tests/models/test_metrics.py (hunk 5 above)
Comment (nirtiac) [reply vulcanai/tests/models/test_metrics.py]: this is just to deal with the sklearn deprecated stuff

**Comment by divyachandran-ds** [comment]
It was for the interactive display part, it was already added in some other
plots so added it similarly with similar time duration
I will update the requirements

**Comment by divyachandran-ds** [comment]
Hi Caitrin

I have updated the requirements

Regards
Divya.

File: vulcanai/tests/plotters/test_visualization.py (hunk 3 above)
Comment (nirtiac) [reply vulcanai/tests/plotters/test_visualization.py]: My comment above is still relevant ^ I believe.

File: vulcanai/tests/models/test_metrics.py [hunk 6]
```diff
@@ -315,11 +313,8 @@ def test_run_test_binary(self, metrics, cnn_class_binary):
         """Test that run_test returns values as expected."""
         num_items = 300
 
-        test_input = torch.Tensor(np.random.randint(0, 2,
-                                                    size=(num_items,
-                                                          *cnn_class_binary.in_dim)))
-        test_target = torch.LongTensor(np.random.randint(0, 2,
-                                                         size=num_items))
+        test_input = torch.Tensor(np.random.randint(0, 2,size=(num_items,*cnn_class_binary.in_dim)))
```
Comment (nirtiac) [inline vulcanai/tests/models/test_metrics.py]: please make sure that every line is within the limit
